verifier/src/lib.rs

```diff
@@ -109,19 +168,27 @@ pub trait Verifier<T>: Debug {
 }
 
 /// An error that occurs during an `and` operation.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct AndError<L, R> {
-    _left: CtOption<L>,
-    _right: CtOption<R>,
+    left: CtOption<L>,
+    right: CtOption<R>,
 }
 
 impl<L, R> AndError<L, R> {
     /// Create a new instance
     pub fn new(left: CtOption<L>, right: CtOption<R>) -> Self {
-        Self {
-            _left: left,
-            _right: right,
-        }
+        Self { left, right }
+    }
+}
+
+impl<L: DisplayableError, R: DisplayableError> DisplayableError for AndError<L, R> {}
+
+impl<L: DisplayableError, R: DisplayableError> Display for AndError<L, R> {
+    fn fmt(&self, f: &mut Formatter) -> core::fmt::Result {
+        f.debug_struct("AndError")
+            .field("left", &format_args!("{:#}", self.left.display()))
+            .field("right", &format_args!("{:#}", self.right.display()))
+            .finish()
```

Comment:
See comments on `OrError`

Reply:
Updated, but don't think it's quite what was being suggested